### PR TITLE
bug(#165963466): Customising data representation in the search functionality

### PR DESCRIPTION
**Pull request by Etomovich**
**Description**
<hr/>

This PR addresses the formatting of the search URL. Currently the search URL is implemented using a PUT method which is changed to a GET method in this PR. It also addresses the fact that searching for articles could yield a large number of articles hence the need to return results that are paginated to save on the lag time the application will take to load all the articles.  The endpoint also supports custom filtering with either `tags`, `author`, `title`, `description`, and `body` all supported as valid filters. This filters are applied as `slugs` in the search URL and the search string is entered as a parameter with the name of `query`. The search URL is as shown below:

`articles/<slug: slug>/search/?query=<search_value>`

For example `tags` would be: `articles/tags/search/?query=<search_value>` e.t.c

**Type of Change**
<hr/>

- [x] bug(a non-breaking change that allows users search for articles).

**Checklist**
<hr/>

- [x] Customising data representation in the search functionality.
- [x] Return paginated search results.
- [x] Add test to show search functionality operations.

**Related Stories**
<hr/>

[Fix search and custom filtering functionality](https://www.pivotaltracker.com/n/projects/2313280/stories/165963466)

### CODE REVIEW

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -801,73 +801,60 @@ class SearchForArticles(generics.ListAPIView):
...
-            query_fields = list(set(query_fields + list(query)))
-
         matching_tags = []
-        if('tags' in search_fields) or (not search_fields):
-            matching_tags = list(Tag.objects.filter(
-                tag__icontains=search_string
-            ))
-        if matching_tags:
-            count = 0
+        matching_tags = list(Tag.objects.filter(
```
Comment (abulojoshua1) [inline authors/apps/articles/views.py:839]: @Etomovich, you do not need to convert this to a list, you can work with the queryset as is.

File: authors/apps/articles/views.py [hunk 2]
```diff
@@ -801,73 +801,60 @@ class SearchForArticles(generics.ListAPIView):
...
+            tag__icontains=search_string
+        ))
+        if matching_tags and slug == "tags":
             for item in matching_tags:
                 query = Article.objects.filter(
-                    tags__id__icontains=item.id
+                    tags__id__icontains=item.id, published=True,
+                    activated=True
+                )
+                payload = list(
```
Comment (abulojoshua1) [inline authors/apps/articles/views.py:848]: Same thing here.

**Review by abulojoshua1 — APPROVED** [review]
Mostly looks good.

File: authors/apps/articles/views.py (hunk 1 above)
Comment (Etomovich) [reply authors/apps/articles/views.py:839]: > @Etomovich, you do not need to convert this to a list, you can work with the queryset as is.

Noted

File: authors/apps/articles/views.py (hunk 2 above)
Comment (Etomovich) [reply authors/apps/articles/views.py:848]: > Same thing here.

Noted too.

**Pull request merged**